### PR TITLE
[*] BO : add seo/sharing social field to new product form

### PR DESCRIPTION
**Pull request by kelu95**

### CODE REVIEW

**Comment by inetbiz** [comment]
I am against doing this. There is already schema vocabulary on the page. Google and Bing also use the vocabulary to eliminate duplicate entries in their database. With more and more social shares being backlinks, this would not be advantageous unless you were setting up google interactive post via API and needed a default sharing message.

**Comment by kelu95** [comment]
Sorry this is not in the roadmap...

File: classes/Product.php [hunk 1]
```diff
@@ -151,6 +151,12 @@ class ProductCore extends ObjectModel
     /** @var string Meta tag title */
     public $meta_title;
 
+    /** @var string Socail sharing description */
```
Comment (PrestaEdit) [inline classes/Product.php]: "Socail" => "Social"

File: classes/Product.php [hunk 2]
```diff
@@ -151,6 +151,12 @@ class ProductCore extends ObjectModel
...
+    public $social_sharing_description;
+
+    /** @var string Socail sharing title */
```
Comment (PrestaEdit) [inline classes/Product.php]: "Socail" => "Social"

File: install-dev/data/db_structure.sql [hunk 3]
```diff
@@ -1610,6 +1610,8 @@ CREATE TABLE `PREFIX_product_lang` (
   `name` varchar(128) NOT NULL,
   `available_now` varchar(255) DEFAULT NULL,
   `available_later` varchar(255) DEFAULT NULL,
+  `social_sharing_title` varchar(255) DEFAULT NULL,
```
Comment (PrestaEdit) [inline install-dev/data/db_structure.sql]: Why not a varchar(128), like the controller is max size ?

File: install-dev/upgrade/sql/1.7.0.0.sql [hunk 4]
```diff
@@ -29,4 +29,7 @@ ALTER TABLE  `PREFIX_product` ADD  `isbn` VARCHAR( 13 ) NULL DEFAULT NULL;
 ALTER TABLE  `PREFIX_order_detail` ADD  `product_isbn` VARCHAR( 13 ) NULL DEFAULT NULL;
 ALTER TABLE  `PREFIX_product_attribute` ADD  `isbn` VARCHAR( 13 ) NULL DEFAULT NULL;
 ALTER TABLE  `PREFIX_stock` ADD  `isbn` VARCHAR( 13 ) NULL DEFAULT NULL;
-ALTER TABLE  `PREFIX_supply_order_detail` ADD  `isbn` VARCHAR( 13 ) NULL DEFAULT NULL;
\ No newline at end of file
+ALTER TABLE  `PREFIX_supply_order_detail` ADD  `isbn` VARCHAR( 13 ) NULL DEFAULT NULL;
+
+ALTER TABLE  `PREFIX_product_lang` ADD  `social_sharing_title` VARCHAR( 255 ) NOT NULL;
```
Comment (PrestaEdit) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Same as before (128/255)

**Comment by kelu95** [comment]
I made some update, thanks PrestaEdit

**Comment by PrestaEdit** [comment]
:+1:

**Comment by julienbourdeau** [comment]
> more and more social shares being backlinks, this would not be advantageous

@inetbiz I'm not sure I understood. Why is it disadvantageous ?

**Comment by inetbiz** [comment]
@julienbourdeau alternate titles in social shares creates unambiguous semantics. An attempt to game the system. That will ultimately fail and lead to penalties from search engines. The connection between say twitter and google now relate the extracted entities and compares them to the source of the share.

Google for instance, would not award a high level of certainty to any altered social backlink because the product schema does not match and reduce the chances of being rewarded a product rich snippet. Snippets in SERP are **not** automatic.

Create good product titles and you won't have to worry about this unneeded feature in the back-office. And lastly, keep it simple. The existing product title is exactly what a store owner needs to focus on content creation.

**Comment by inetbiz** [comment]
Mr. David Amerland has published 5 books about SEO, Structured Data. He consults with major corporations and is someone I have interviewed, personally several times. It's not just my professional opinion at play, here.

**Comment by radusi** [comment]
+1 @inetbiz . it is a questionable practice at most, if not "grey hat" . cannot lead to long term healthy results.

**Comment by xGouley** [comment]
Hello,
Your suggestion has not been dropped. I need to merge this PR now to continue working on the develop branch, but I already transferred your opinion to our SEO expert. I'm waiting for a decision from my expert and can rollback if needed.

Best regards.

**Comment by inetbiz** [comment]
@xGouley Grey Hat SEO isn't even professional SEO. You are harming a sites domain authority with ambiguity of entity extraction. They may reach out to Mr. Amerland on G+, anytime. Again, he consults for fortune 500.

**Comment by kelu95** [comment]
Hi, 
after intebiz recommandations, we decide to cancel this PR, thanks to him.
Also, this features already exists on SEO expert module.

Best regards

**Comment by inetbiz** [comment]
@kelu95 I hope you understand? My sentiment was only for the good of the project.

**Pull request merged**